app/Helpers/CustomHelper.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+namespace App\Helpers;
+
+class CustomHelper
+{
+
+   public static function formatSizeUnits($bytes)
+    {
+        if ($bytes >= 1073741824)
+        {
+            $bytes = number_format($bytes / 1073741824, 2) . ' GB';
+        }
+        elseif ($bytes >= 1048576)
+        {
+            $bytes = number_format($bytes / 1048576, 2) . ' MB';
+        }
+        elseif ($bytes >= 1024)
+        {
+            $bytes = number_format($bytes / 1024, 2) . ' KB';
+        }
+        elseif ($bytes > 1)
+        {
+            $bytes = $bytes . ' bytes';
+        }
+        elseif ($bytes == 1)
+        {
+            $bytes = $bytes . ' byte';
+        }
+        else
+        {
+            $bytes = '0 bytes';
+        }
```

Comment:
Nothing needed here - this function works well and I like the attention to detail shown distinguishing between singular and plural in the case where a file is 1 single byte! 

That said - a few comments.

**A note on refactoring**
I recently watched a [video on declarative vs imperative coding](https://laracasts.com/series/guest-spotlight/episodes/4). The presenter discussed "lookup tables" in code - a practice of separating your code into "things that change" and "things that happen". 

Your function here is perfectly good - it works fine, is understandable and is almost exactly what I wrote on my first pass at this function. But it's not hugely elegant. A long set of `elseif`s like this is a good indicator that this function could be refactored at some point. 

**My version**
My version (below, from the GenericHelper class) is also not perfect - as you noted, it's not easy to understand what's happening - why is the `for` loop there? What's happening to the $bytes variable? But it does do a couple of things:

 - it puts all the strings (the things that change) together in one place. So if we ever wanted to change it, e.g. to use the longform "bytes, kilobytes etc", it's easy to do. 
 - it has only a single path through the code,  so it's (potentially) easier to debug if something goes wrong. 






```
public static function fileSizeToHuman($bytes)
    {
        $units = ['B', 'KB', 'MB', 'GB', 'TB' ,'PB'];

        for($i = 0; $bytes > 1024; $i++){
            $bytes /= 1024;
        }

        return round($bytes, 2) . ' ' . $units[$i];
    }
```